cmd/ttn-lw-cli/commands/use.go

```diff
@@ -0,0 +1,147 @@
+		Use:               "use [host]",
+		Short:             "Use",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errNoHost
+			}
+			insecure, _ := cmd.Flags().GetBool("insecure")
+			ca, _ := cmd.Flags().GetBool("ca")
+			user, _ := cmd.Flags().GetBool("user")
+			overwrite, _ := cmd.Flags().GetBool("overwrite")
+
+			hash := func(in string) string {
+				h := md5.New()
+				io.Copy(h, strings.NewReader(in))
+				return hex.EncodeToString(h.Sum(nil))
+			}
+
+			destPath := func(base string, user bool, overwrite bool) (string, error) {
+				fileName := base
+				if user {
+					dir, err := os.UserConfigDir()
+					if err != nil {
+						return "", err
+					}
+					fileName = filepath.Join(dir, base)
+				}
+
+				_, err := os.Stat(fileName)
+				if os.IsNotExist(err) || overwrite {
+					return fileName, nil
+				}
+
+				logger.Warnf("%s already exists. Use --overwrite", fileName)
+				return "", errFileExists.WithAttributes("file", fileName)
+			}
+
+			caFileName := func(host string) string {
+				return fmt.Sprintf("ca.%s.pem", hash(host)[:6])
+			}
+
+			host := args[0]
+			conf := MakeDefaultConfig(host, insecure)
+			conf.CredentialsID = host
+
+			// Get CA certificate from server
+			if !insecure && ca {
+				logger.Infof("Will retrieve certificate from %s", conf.NetworkServerGRPCAddress)
+				caFile, err := destPath(caFileName(host), user, overwrite)
+				if err != nil {
+					return err
+				}
+
+				conn, err := tls.Dial("tcp", conf.NetworkServerGRPCAddress, &tls.Config{InsecureSkipVerify: true})
+				if err != nil {
+					return err
+				}
+				defer conn.Close()
+
+				var buf bytes.Buffer
+				for _, cert := range conn.ConnectionState().PeerCertificates {
+					if !cert.IsCA {
+						continue
+					}
+					if err = pem.Encode(&buf, &pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw}); err != nil {
+						logger.Warnf("Could not retrieve certificate: %s", err)
+					}
+				}
+				if err = ioutil.WriteFile(caFile, buf.Bytes(), 0644); err != nil {
+					return errFailWrite.WithCause(err).WithAttributes("file", caFile)
+				}
+				logger.Infof("CA file for %s written in %s", host, caFile)
+				abs, err := filepath.Abs(caFile)
+				if err != nil {
+					return err
+				}
+				conf.CA = abs
+			}
+
+			b, err := yaml.Marshal(conf)
+			if err != nil {
+				return err
+			}
+
+			configFile, err := destPath(configFileName, user, overwrite)
+			if err != nil {
+				return err
+			}
+
+			if err = ioutil.WriteFile(configFile, b, 0644); err != nil {
+				return errFailWrite.WithCause(err).WithAttributes("file", configFile)
+			}
+			logger.Infof("Config file for %s written in %s", host, configFile)
+			return nil
+		},
+	}
+)
+
+func init() {
+	useCommand.Flags().Bool("insecure", defaultInsecure, "Connect without TLS")
+	useCommand.Flags().String("host", defaultClusterHost, "Server host name")
+	useCommand.Flags().Bool("ca", false, "Connect to server and retrieve CA")
```

Comment:
Can we rename this flag to `--fetch-ca` or something? I think `--ca` is confusing.